pkg/util/storage.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"crypto/x509"
+	"fmt"
+	"net/url"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"github.com/operator-framework/rukpak/internal/storage"
+	"github.com/operator-framework/rukpak/internal/util"
+	pkgstorage "github.com/operator-framework/rukpak/pkg/storage"
+)
+
+const DefaultBundleCacheDir = "/var/cache/bundles"
+
+func NewBundleStorage(mgr ctrl.Manager, provisionerStorageDirectory, httpExternalAddr, bundleCAFile, bearerToken string) (pkgstorage.Storage, error) {
```

Comment:
IMO, this should go in the `pkg/storage` package.

Reply:
I'm trying to avoid the `import cycle` like this.  
```
package github.com/operator-framework/rukpak/cmd/core
	imports github.com/operator-framework/rukpak/internal/finalizer
	imports github.com/operator-framework/rukpak/pkg/storage
	imports github.com/operator-framework/rukpak/internal/storage
	imports github.com/operator-framework/rukpak/pkg/storage: import cycle not allowed
```

`NewBundleStorage` refers to both internal/storage and pkg/storage.  It causes the import cycle.  What is the good way to avoid it?  @joelanford

Reply:
oh, I see.  I'll move `pkg/util/storage.go` to `pkg/storage/util/util.go`